### PR TITLE
Add constant interpolated strings

### PR DESCRIPTION
**Pull request by kevinsun-dev**
Add proposal spec for constant interpolated strings

### CODE REVIEW

File: proposals/constant_interpolated_strings.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+*  Identity conversions
+*  Numeric conversions
+*  Enumeration conversions
+*  Constant expression conversions
+*  Implicit and explicit reference conversions, provided that the source of the conversions is a constant expression that evaluates to the null value.
+
+Other conversions including boxing, unboxing and implicit reference conversions of non-null values are not permitted in constant expressions. For example:
+```csharp
+class C {
+    const object i = 5;         // error: boxing conversion not permitted
```
Comment (jaredpar) [inline proposals/constant_interpolated_strings.md:78]:
```suggestion
class C 
{
    const object i = 5;         // error: boxing conversion not permitted
```

Should keep formatting consistent for all samples in the doc

File: proposals/constant_interpolated_strings.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+A *constant_expression* is an expression that can be fully evaluated at compile-time.
+
+```antlr
+constant_expression
+    : expression
+    ;
+```
+
+A constant expression must be the `null` literal or a value with one of  the following types: `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `char`, `float`, `double`, `decimal`, `bool`, `object`, `string`, or any enumeration type. Only the following constructs are permitted in constant expressions:
```
Comment (jaredpar) [inline proposals/constant_interpolated_strings.md:51]: Looks like this is coming mostly from the C# spec, should attribute that in the writing here. Basically talk about which section of which spec version you got this from.

File: proposals/constant_interpolated_strings.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+*  References to `const` parameters or local variables
+*  Parenthesized sub-expressions, which are themselves constant expressions.
+*  Cast expressions, provided the target type is one of the types listed above.
+*  `checked` and `unchecked` expressions
+*  Default value expressions
+*  Nameof expressions
+*  The predefined `+`, `-`, `!`, and `~` unary operators.
+*  The predefined `+`, `-`, `*`, `/`, `%`, `<<`, `>>`, `&`, `|`, `^`, `&&`, `||`, `==`, `!=`, `<`, `>`, `<=`, and `>=` binary operators, provided each operand is of a type listed above.
+*  The `?:` conditional operator.
+*  *Interpolated strings `${}`, provided that all components are constant expressions of type `string` and all interpolated components lack format specifiers.*
```
Comment (jaredpar) [inline proposals/constant_interpolated_strings.md:65]: Interpolated components have three parts: expression, alignment and format specifiers. The alignment portion isn't mentioned here. Is the intent to support it? My suggestion would be no for now unless we can find compelling reasons to do so.  

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated

Comment (kevinsun-dev) [reply proposals/constant_interpolated_strings.md]: Thanks, added further restrictions on alignment specifiers as well.

File: proposals/constant_interpolated_strings.md (hunk 2 above)
Comment (kevinsun-dev) [reply proposals/constant_interpolated_strings.md:51]: Added direct link to section of current spec.

**Pull request merged**